src/routes/v1/gateway.ts

```diff
@@ -117,6 +118,16 @@ async function handleMessage(message, ws, user, socketId) {
                 let Quarks = db.getQuarks();
                 try {
                     let quark = await Quarks.findOne({_id: event.split("_")[1], members: user._id});
+                    if (ConstantID_DMvQuark.equals(event.split("_")[1]) || event.split("_")[1] === "dm") {
+                        quark = {
+                            name: "Direct Messages",
+                            _id: ConstantID_DMvQuark,
+                            members: [user._id],
+                            invite: "direct messages", // Impossible invite
+                            owners: [ ConstantID_SystemUser ],
+                            channels: []
+                        }

```

Comment:
You paste this one billion trillion times, perhaps it should be a constant stored somewhere

Reply:
yeah ive been thinking about this

Reply:
never had an original thought in my life

Reply:
then i decided im lazy and didnt make it a constant because the user id and channels keep changing